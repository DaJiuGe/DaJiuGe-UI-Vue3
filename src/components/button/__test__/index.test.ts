import { describe, expect, it } from 'vitest'
import { mount } from '@vue/test-utils'
import SuButton from '../index.vue'

describe('Checkbox', () => {
  it('should toggle checked when click', async () => {
    const wrapper = mount(SuButton)
    expect(wrapper.find('.su-button').exists()).toBeTruthy()
    await wrapper.trigger('click')
    expect(wrapper.html()).toMatchSnapshot()
    expect(wrapper.find('.su-button.is-round').exists()).toBeTruthy()
  })
})
