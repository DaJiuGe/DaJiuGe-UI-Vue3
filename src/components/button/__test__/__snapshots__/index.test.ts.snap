// Vitest Snapshot v1

exports[`Checkbox > should toggle checked when click 1`] = `
"<button class=\\"su-button su-button--default is-round\\" data-v-c89ffb08=\\"\\">
  <!--v-if-->
  <!--v-if-->
</button>"
`;
